driver/src/main/java/org/neo4j/driver/internal/async/connection/BoltProtocolUtil.java

```diff
@@ -20,32 +20,35 @@
 
 import io.netty.buffer.ByteBuf;
 
+import org.neo4j.driver.internal.messaging.BoltProtocolVersion;
 import org.neo4j.driver.internal.messaging.v1.BoltProtocolV1;
 import org.neo4j.driver.internal.messaging.v2.BoltProtocolV2;
 import org.neo4j.driver.internal.messaging.v3.BoltProtocolV3;
 import org.neo4j.driver.internal.messaging.v4.BoltProtocolV4;
+import org.neo4j.driver.internal.messaging.v41.BoltProtocolV41;
 
 import static io.netty.buffer.Unpooled.copyInt;
 import static io.netty.buffer.Unpooled.unreleasableBuffer;
 import static java.lang.Integer.toHexString;
 
 public final class BoltProtocolUtil
 {
-    public static final int HTTP = 1213486160; //== 0x48545450 == "HTTP"
+    // Represents the `TP` part of `HTTP` (0x48545450) which can be returned if connected to HTTP port
+    public static final BoltProtocolVersion HTTP = new BoltProtocolVersion(80, 84 );
```

Comment:
```suggestion
    public static final BoltProtocolVersion HTTP = new BoltProtocolVersion( 80, 84 );
```
Also it feels at least a little bit hacky to use the `BoltProtocolVersion` for a `T,P`.
Maybe only as an inner class or similar of `HandshakeHandler` or a method in `BoltProtocolVersion` like `isHttp`.

Reply:
move this into a `isHttp()` method on `BoltProtocolVersion`